L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -109,19 +161,139 @@ namespace trackFindingTracklet {
       Handle<TTTrackRefMap> handleTTTrackRefMap;
       iEvent.getByToken<TTTrackRefMap>(edGetTokenTTTrackRefMap_, handleTTTrackRefMap);
       const TTTrackRefMap& ttTrackRefMap = *handleTTTrackRefMap.product();
-      // perform KFout emulation and fill accepted and lost
-      // StreamsTrack is a vector of StreamTrack which is a vector of FrameTrack which is a pair of an edm::Ref<TTTrack> and std::bitset<64>
-      // the std::bitset<64> are the frames of an emp link
-      // the edm::Ref<TTTrack> is used to meassure tracking efficiency
-      // your input streamsTracks contain edm::Ref<TTTrack> to KF input TTTracks
-      // use ttTrackRefMap to lookup edm::Ref<TTTrack> of KF output TTTracks, that allows us to meassure the tracking efficiency after the KFout block
-      // your output frames belong to either only one TTTrack or to two, in the later case chose any edm::Ref<TTTrack> of the two
-    }
+      // 18 Output Links (First Vector) each has a vector of tracks per event (second vector) each track is 3 32 bit TTBV partial tracks 
+      vector<vector<TTBV>> SortedPartialTracks(setup_->numRegions() * setup_->tfpNumChannel(),vector<TTBV>(0));   
+      StreamsTrack OutputStreamsTracks(setup_->numRegions() * setup_->tfpNumChannel());
+
+      for (int iLink = 0; iLink < (int)streamsTracks.size(); iLink++ ){
+
+        for (int iTrack = 0; iTrack < (int)streamsTracks[iLink].size(); iTrack++ ){
+          auto track = streamsTracks[iLink].at(iTrack);
```

Comment:
Please use const auto& if you are not manipulating track

Reply:
done